### PR TITLE
Added a feature AllOptionGroup

### PR DESCRIPTION
**Pull request by roopahc**
Fixes: https://github.com/click-contrib/click-option-group/issues/12

### CODE REVIEW

File: click_option_group/_core.py [hunk 1]
```diff
@@ -351,3 +351,26 @@ def handle_parse_result(self, option: GroupedOption, ctx: click.Context, opts: d
...
+
+
+class AllOptionGroup(OptionGroup):
+    """Option group with required all/none options of this group
+
+    `AllOptionGroup` defines the behavior:
+    - All options from the group must be set or None must be set.
+    """
+
+    @property
```
Comment (espdev) [inline click_option_group/_core.py:363]: I think we need to add `required` and `hidden` option attributes to the list of forbidden attributes for this class:
https://github.com/click-contrib/click-option-group/blob/a47e2216bc5c006824ddc0602662b2db03c714bd/click_option_group/_core.py#L286-L288

Comment (roopahc) [reply click_option_group/_core.py:363]: @espdev, Updated

**Comment by roopahc** [comment]
@espdev would you please help me merge this PR? Please let me know what is the process to release this package.

**Comment by espdev** [comment]
@roopahc thank you! I will make a new release soon. I will let you know here.

**Comment by espdev** [comment]
@roopahc done
https://pypi.org/project/click-option-group/0.5.0/

**Comment by roopahc** [comment]
Thank you very much @espdev  :)

**Pull request merged**